### DISCUSSION TITLE
Encoder only increments value

### DISCUSSION THREAD

**Discussion by rafaelfcsouza** [tttapa/Control-Surface#410] [Q&A]
Hello, I'm starting to adventure myself in the DIY controller world, starting with an encoder example provided in the github repo:

```
#include <Encoder.h> // Include the Encoder library.
// This must be done before the Control Surface library.
#include <Control_Surface.h> // Include the Control Surface library

// Instantiate a MIDI over USB interface.
HairlessMIDI_Interface midi;

// Instantiate a CCRotaryEncoder object
CCRotaryEncoder enc = {
  {2, 3},       // pins
  MCU::V_POT_1, // MIDI address (CC number + optional channel)
  1,            // optional multiplier if the control isn't fast enough
};

void setup() {
  // Select the correct relative MIDI CC mode.
  // Options:
  //   - TWOS_COMPLEMENT (default)
  //   - BINARY_OFFSET
  //   - SIGN_MAGNITUDE
  //   - NEXT_ADDRESS
  // Aliases:
  //   - REAPER_RELATIVE_1
  //   - REAPER_RELATIVE_2
  //   - REAPER_RELATIVE_3
  //   - TRACKTION_RELATIVE
  //   - MACKIE_CONTROL_RELATIVE
  //   - KORG_KONTROL_INC_DEC_1
  RelativeCCSender::setMode(relativeCCmode::MACKIE_CONTROL_RELATIVE);
  Control_Surface.begin(); // Initialize Control Surface
}

void loop() {
  Control_Surface.loop(); // Update the Control Surface
}
```

I'm using Hairless Serial to MIDI because according to the documentation, Arduino Uno do not support MIDI, unless I change the Firmware.

Hairless debug:
```
Starting MIDI<->Serial Bridge...
Opening serial port '/dev/cu.(...)'...
Opening MIDI Out port #0
+198.52 - Serial In: Ch 1: Controller 16 value 1
+198.52 - Serial In: Ch 1: Controller 16 value 1
+198.536 - Serial In: Ch 1: Controller 16 value 1
.
.
.
```

I can connect to Ableton, but I just see the knob of selected parameter incrementing regardless of which direction I turn the knob. 

Pictures of the proto:
![IMG_0088](https://user-images.githubusercontent.com/28117774/109385713-2ef49180-78f6-11eb-98d9-5160d17bfb48.jpg)

Could some one help me?

**Comment by tttapa** [comment]
Issues like this are often hardware-related. Have you tried the examples that come with the Encoder.h library?

The Vcc line in your circuit is not connected to your encoder, the breadboard supply buses are not connected in the middle (this is indicated by the gap in the red and blue lines).

**Reply by rafaelfcsouza** [reply]
omg, such newbie, this was the issue :/ thank you very much!
